addons/event_stand/models/event_stand_slot.py

```diff
@@ -0,0 +1,144 @@
+            domain = [
+                ('id', '!=', slot.id),
+                ('event_stand_id', '=', slot.event_stand_id.id),
+                '|', '|',
+                '&', ('date_from', '<', slot.date_from), ('date_to', '>', slot.date_from),
+                '&', ('date_from', '<', slot.date_to), ('date_to', '>', slot.date_to),
+                '&', ('date_from', '<', slot.date_from), ('date_to', '>', slot.date_to)
+            ]
+            if self.search_count(domain) > 0:
+                raise ValidationError(_('You can not have overlapping slots.'))
+
+    @api.depends('slot_registration_ids')
+    def _compute_slot_registration_count(self):
+        for slot in self:
+            slot.slot_registration_count = len(slot.slot_registration_ids)
+
+    def name_get(self):
+        result = []
+        tz = pytz.timezone(self.env.user.tz)
+        for slot in self:
+            name = '%s \U0001F852 %s' % (
+                slot.date_from.astimezone(tz).replace(tzinfo=None),
+                slot.date_to.astimezone(tz).replace(tzinfo=None) if slot.date_from.date() != slot.date_to.date() else slot.date_to.astimezone(tz).time()
+            )
+            result.append([slot.id, name])
+        return result
+        # return [(slot.id, '%s \U0001F852 %s' % (slot.date_from, slot.date_to if slot.date_from.date() != slot.date_to.date() else slot.date_to.time())) for slot in self]
+
+    # TODO: This is considering the registrations are contiguous (and they should be !)
+    def _get_stand_multiline_description(self):
+        from_date = min(self.mapped('date_from'))
+        to_date = max(self.mapped('date_to'))
+        tz = pytz.timezone(self.env.user.tz)
+        duration_display = '%s \U0001F852 %s' % (
+            from_date.astimezone(tz).replace(tzinfo=None),
+            to_date.astimezone(tz).replace(tzinfo=None) if from_date.date() != to_date.date() else to_date.astimezone(tz).time()
+        )
+        return '%s - %s\n%s' % (self.event_stand_id.name, duration_display, self.event_id.display_name)
+
+    def action_set_available(self):
+        self.write({'state': 'available'})
+
+    def action_confirm(self, registration_id):
+        if not self.is_available():
+            raise ValidationError(_('At least one slot is unavailable, please modify your registration and retry'))
+        self.write({
+            'state': 'unavailable',
+            'partner_id': registration_id.partner_id,
+            'name': registration_id.name,
+            'email': registration_id.email,
+            'phone': registration_id.phone,
+            'mobile': registration_id.mobile,
+        })
+
+    def action_reserve(self):
+        self.write({
+            'state': 'reserved',
+            'partner_id': self.env.user.partner_id.id,
+        })
+
+    def action_create_new_registration(self):
+        self.ensure_one()
+        return self._get_create_new_registration_view()
+
+    def action_create_new_registrations(self):
```

Comment:
You could probably have a single method and in _get_create_new_registration_view handle singleton recordset / multi record set directly to simplify a bit code.